R/epiparam_utils.R

```diff
@@ -0,0 +1,447 @@
+#' Converts rows of an epiparam object into an epidist object
+#'
+#' @description Can convert the entries (rows) of an epiparam object to one or
+#' list of several epidist objects. This allows subsetting of epiparam records
+#' into a class that can then be used to to calculate the cdf, random number
+#' generation and other methods of the epidist class
+#' (see `?epidist_distribution_functions`).
+#'
+#' @param x A epiparam object
+#'
+#' @return An epidist object or a list of epidist objects
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#'   eparam <- epiparam()
+#'   as_epidist(eparam[1, ])
+#' }
+as_epidist <- function(x) {
+
+  # check input
+  validate_epiparam(x)
+
+  # convert each epiparam row into an epidist object
+  out <- apply(x, 1, function(y) {
+    make_epidist(x = y)
+  })
+
+  # remove names of list
+  names(out) <- NULL
+
+  # unpack from list if only one epidist
+  if (length(out) == 1) out <- out[[1]]
+
+  # return list of epidist objects
+  out
+}
+
+#' Creates an `epidist` object from a list of input from an `epiparam` object
+#'
+#' @description Unpacks list of inputs from an epiparam object into the epidist
+#' helper, including the parameters and uncertainty from the correct type of
+#' probability distribution
+#'
+#' @param x List of data to be used to construct an epidist object
+#'
+#' @return An `epidist` object
+#' @keywords internal
+make_epidist <- function(x) {
+
+  # determine parameters
+  if (x$prob_distribution %in% c("gamma", "weibull")) {
+    parameters <- c(shape = x$shape, scale = x$scale)
+    uncertainty <- list(
+      shape = create_epidist_uncertainty(
+        ci_limits = x$shape_ci_limits,
+        ci = x$shape_ci,
+        ci_type =  ifelse(
+          test = x$inference_method == "mle",
+          yes = "confidence interval",
+          no = "credible interval"
+        )
+      ),
+      scale = create_epidist_uncertainty(
+        ci_limits = x$scale_ci_limits,
+        ci = x$scale_ci,
+        ci_type = ifelse(
+          test = x$inference_method == "mle",
+          yes = "confidence interval",
+          no = "credible interval"
+        )
+      )
+    )
+  } else if (x$prob_distribution %in% "lognormal") {
+    parameters <- c(meanlog = x$meanlog, sdlog = x$sdlog)
+    uncertainty <- list(
+      meanlog = create_epidist_uncertainty(
+        ci_limits = x$meanlog_ci_limits,
+        ci = x$meanlog_ci,
+        ci_type =  ifelse(
+          test = x$inference_method == "mle",
+          yes = "confidence interval",
+          no = "credible interval"
+        )
+      ),
+      sdlog = create_epidist_uncertainty(
+        ci_limits = x$sdlog_ci_limits,
+        ci = x$sdlog_ci,
+        ci_type = ifelse(
+          test = x$inference_method == "mle",
+          yes = "confidence interval",
+          no = "credible interval"
+        )
+      )
+    )
+  } else if (x$prob_distribution %in% "negative_binomial") {
```

Comment:
might want to consider using R abbreviations, i.e. "lnorm", "nbinom" etc.

Reply:
This will be changed in an upcoming pull request. The discussion of naming conventions is here #86